Automattic/jetpack
# Social Notes: Add options and UI for link formatting

Adds options and UI for link formatting options for Jetpack Social Notes

## Proposed changes:

* Add `social_notes_config` option to Jetpack social options
* Add three options for Jetpack social config

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to Jetpack Social admin page
* If Social Notes is disabled, confirm that you do not see the Notes config options 
* Turn ON Social Notes
* Confirm that you see the config options
* Confirm that "Append post link" option is enabled by default.
* Turn OFF "Append post link" option
* Confirm that other two options are hidden
* Change different options
* Reload the page to confirm that the options are persisted.

1. When Social Notes is disabled
    <img width="784" alt="Screenshot 2024-04-01 at 6 04 45 PM" src="https://github.com/Automattic/jetpack/assets/18226415/9fe8d5f9-e37e-4ee6-86d6-76ff610273e6">

2. When Social Notes is enabled, it shows the default values
    <img width="859" alt="Screenshot 2024-04-01 at 6 04 57 PM" src="https://github.com/Automattic/jetpack/assets/18226415/bcc342c5-8a3d-4609-831f-adfea15fb237">

3. When "Append post link" is disabled, it hides the other two options
    <img width="807" alt="Screenshot 2024-04-01 at 6 05 09 PM" src="https://github.com/Automattic/jetpack/assets/18226415/d658e814-19b9-41c6-aacd-f4fdff34f768">

4. When all options are enabled
    <img width="815" alt="Screenshot 2024-04-01 at 6 05 23 PM" src="https://github.com/Automattic/jetpack/assets/18226415/9c4c2375-e0e6-4125-bf12-51740775cc68">

5. Small screen view
    <img width="398" alt="Screenshot 2024-04-01 at 6 05 33 PM" src="https://github.com/Automattic/jetpack/assets/18226415/fc96b6fa-0b35-4054-a876-efcf44f2744b">
